pkg/controller/dnsendpoint/dnsendpoint_controller.go

```diff
@@ -145,10 +179,34 @@ func (r *ReconcileDNSEndpoint) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, nil
 	}
 
-	domain := instance.Spec.Zone
-	dnsLog = dnsLog.WithField("domain", domain)
+	fullDomain := instance.Spec.Zone
+
+	dnsLog = dnsLog.WithField("domain", fullDomain)
 
-	if !r.nameServerScraper.HasBeenScraped(domain) {
+	var rootDomain string
+	var currentNameServers sets.String
+
+	var nsTool nameServerTool
+
+	for i, nst := range r.nameServerTools {
+		rootDomain, currentNameServers = nst.scraper.GetEndpoint(fullDomain)
+		if rootDomain != "" {
+			nsTool = r.nameServerTools[i]
+			break
+		}
+	}
+
+	if rootDomain == "" {
+		dnsLog.WithField("domain", fullDomain).Error("no scraper for domain found, skipping reconcile")
+		_, err := updateMissingDomainClientCondition(r.Client, dnsLog, instance, true)
```

Comment:
This condition will not get set in the case where the managed DNS was removed completely from HiveConfig, since there will be no reconciler running. I think that we want to keep the reconciler running now even when managed DNS is disabled.

Reply:
definitely. will modify.